### ISSUE TITLE
DisableRedirectHandling in JenkinsHttpClient

### ISSUE THREAD

**Issue by amcginn** [jenkinsci/java-client-api#235]
When calling a GET for job/<job>/<build#>/toggleLogKeep, I get an exception thrown:
org.apache.http.client.CircularRedirectException: Circular redirect to 'https://<server>/job/<job>/<build#>/toggleLogKeep/api/'

If I edit the source code in JenkinsHttpClient.addAuthentication(HttpClientBuilder builder, URI uri, String username, String password) to add HttpClientBuilder.disableRedirectHandling(), this prevents the Exception from being thrown and allows me to toggle the "Keep build forever" functionality.

**Comment by khmarbaise** [comment]
But that means your environment makes such redirection...and not the api. The api only follows the redirect...Apart from that you could also offer a patch to improve that handling in java-client-api ?

**Comment by amcginn** [comment]
True, the Jenkins API is causing the cyclical redirect. Updating the HttpClientBuilder in the java-client-api to ignore redirects will prevent the exception from being thrown in these situations. Opened #266 for this fix.
